src/general/MathUtils.cs

```diff
@@ -74,4 +74,14 @@ public static float Lerp(float from, float to, float weight, float tolerance = M
 
         return Mathf.Lerp(from, to, weight);
     }
+
+    /// <summary>
+    ///   Standard modulo for negative values in C# produces negative results.
+    ///   This function returns modulo values between 0 and mod-1.
+    /// </summary>
+    /// <returns>The positive modulo</returns>
+    public static int PositiveModulo(this int val, int mod)
+    {
+        return (val % mod + mod) % mod;
```

Comment:
What is this supposed to do? From my experiments `5.PositiveModulo(3)` returns 2, makes sense, but `(-5).PositiveModulo(3)` returns 1?

Reply:
This is the normal mathematical modulo(at least I think I learned that in school). For negative numbers add the mod until it is in range [0;mod). For positive numbers subtract the mod until it is in range [0;mod)
[link](https://torstencurdt.com/tech/posts/modulo-of-negative-numbers/) e.g. Rust would return positive modulo for %.

Reply:
This article also shows this code:
```
int mod(a, b) {
  c = a % b
  return (c < 0) ? c + b : c
}
```

I think this is way readable and should be more efficient because modulo gets used only once